### ISSUE TITLE
API filtering for api/20/project/[PROJECT]/executions inconsistently returning less than the max specified.

### ISSUE THREAD

**Issue by KaushikKoirala** [rundeck/rundeck#5150]
**Describe the bug**
When I submit a request to an the project/[PROJECT]/executions endpoint with query parameters like so:
`/api/20/project/[PROJECT]/executions?olderFilter=2y&max=10`
I will get an API response with the following attributes: 
`"paging": {
        "count": 1,
        "total": 87654,
        "offset": 0,
        "max": 10
    },`
If I set a max of 10, I should be getting a count of 10 executions in the response list, but I only get back 1 execution in the response. Additionally, the actual count each time is unpredictable. So if I make the same request to the API with an max of 400, I will get 251 executions in the response rather than the 400 I indicated I wanted. There doesn't seem to be any pattern in the counts that it is returning based on the max that I input as a query parameter.

**My Rundeck detail**

* Rundeck version: 2.9.3 (API Reference Version 20)
* install type: [launcher]
* OS Name/version: [ubuntu 14.04 LTS]
* DB Type/version: [postgres ]

**To Reproduce**
` curl -s -H x-rundeck-auth-token:$RD_TOKEN /api/20/project/[project]/executions?olderFilter=2y&max=10`
Observe that response has less executions than the max specified although the total is far higher. 

**Expected behavior**
If I pass in a 400 as the value for query parameter `max` I expect 400 executions in the API response. 

**Additional context**
I have searched through similar issues and bug reports. Some indicated similar, although not exactly the same problem when using a `Content-Type` header in their request. I am using an `Accept` header specifying I want my response as `application/json` but even when removing that header and trying the filtered get request, the XML seems to have the same problem of returning less executions than the max specified.

**Comment by MegaDrive68k** [comment]
Hi @KaushikKoirala 

Try with:

`curl --location --request GET "http://yourhost:4440/api/32/job/9f234666-28d8-4cca-98b5-cb05a3b07f89/executions?max=1" --header "Accept: application/json" --header "X-Rundeck-Auth-Token: qPF54dK1EtZa7mwYcYnt7VKkxpucaAWH" --header "Content-Type: application/json"`

Works using 3.0.24 / 3.1.0.

Hope it helps!

**Comment by KaushikKoirala** [comment]
Hi @MegaDrive68k,
Version 32 of the API as you are using is unsupported. Additionally, when I try to reproduce your query I do get what you get. In other words, when I set a max of 25, I get 25 executions and when I set a max of 30 I get 30 executions. The problem still persists when using an `olderFilter` parameter in the request. E.g.,
`http://yourhost:4440/api/32/job/9f234666-28d8-4cca-98b5-cb05a3b07f89/executions?max=25&olderFilter=2y`
This makes sense as olderFilter is not a supported parameter to filter for the `job/[id]/executions` endpoint and I need to use the `project/[id]/executions` endpoint.
